rosbridge_library/src/rosbridge_library/internal/publishers.py

```diff
@@ -105,7 +105,7 @@ def __init__(self, topic, node_handle, msg_type=None, latched_client_id=None, qu
         # without the need of a custom message publisher implementation.
         publisher_qos = QoSProfile(
             depth=queue_size,
-            durability=DurabilityPolicy.TRANSIENT_LOCAL,
+            durability=DurabilityPolicy.VOLATILE,
```

Comment:
Why did you also change the publisher's qos?
```suggestion
            durability=DurabilityPolicy.TRANSIENT_LOCAL,
```